beacon_node/execution_layer/src/engine_api.rs

```diff
@@ -337,16 +337,16 @@ pub struct GetPayloadResponse<T: EthSpec> {
 }
 
 impl<T: EthSpec> GetPayloadResponse<T> {
-    pub fn execution_payload(self) -> ExecutionPayload<T> {
+    pub fn execution_payload(&self) -> ExecutionPayloadRef<T> {
         match self {
             GetPayloadResponse::Merge(response) => {
-                ExecutionPayload::Merge(response.execution_payload)
+                ExecutionPayloadRef::<T>::Merge(&response.execution_payload)
             }
             GetPayloadResponse::Capella(response) => {
-                ExecutionPayload::Capella(response.execution_payload)
+                ExecutionPayloadRef::<T>::Capella(&response.execution_payload)
             }
             GetPayloadResponse::Eip4844(response) => {
-                ExecutionPayload::Eip4844(response.execution_payload)
+                ExecutionPayloadRef::<T>::Eip4844(&response.execution_payload)
             }
```

Comment:
I think we could write this as

```rust
map_get_payload_response_into_execution_payload_ref!(
    response,
    |response, cons| cons(&response.execution_payload)
);
```

With an appropriate `map_into(ExecutionPayloadRef)` on the superstruct declaration

similarly for the owning version we could use `map_into(ExecutionPayload)`

Reply:
Good call! The `map_into` macros expect the owned object and not the reference, but I was able to use `map_into_ref` to map the `Ref` types to one another and then implement this by simply writing `self.to_ref().into()` :)